src/session/SessionStore.ts

```diff
@@ -1,109 +1,66 @@
-            new AccessTokenData(rawAccessTokenData) :
-            undefined;
+        return rawAccessTokenData ? rawAccessTokenData[SessionKeys.AccessToken] : undefined;
     }
 
-    async load(sessionId: string): Promise<Session> {
-
-        let encodedData;
-
-        try {
-            encodedData = await cache.get(sessionId);
-
-        } catch (error) {
-
-            this._logger.error(error.message);
-            throw new Error("Error trying to retrieve from cache");
-        }
-
-        const data = await this.decodeSession(encodedData);
-
-        const accessTokenData = this.getAccessTokenData(data);
-
-        if (accessTokenData && accessTokenData.expiresIn) {
-            this.validateExpiration(accessTokenData.expiresIn);
-        }
-
-        return new Session(sessionId, accessTokenData);
-    }
+    public loadSession = (sessionSecret: string, cookie: string): EitherAsync<Failure, VerifiedSession> => {
 
-    async store(session: Session) {
+        const decodeSession = liftFunctionToAsyncEither<Failure, Session, string>(Encoding.decodeSession);
+        const validateCookieString = liftEitherFunctionToAsyncEither(Cookie.validateCookieString(sessionSecret));
 
-        if (session.id === "") {
-            session.id = await Encoding.generateRandomBytesBase64(this._idOctets);
-            if (!session.accessToken) {
-                session.accessToken = new AccessTokenData({
-                    [SessionKeys.AccessToken]: this.generateExpiry()
-                });
-            }
-        }
+        return liftToAsyncEither<Failure, string>(cookie)
+            .chain(validateCookieString).map(_ => _.sessionId)
+            .chain(this.cache.get)
+            .chain(decodeSession)
+            .chain(session => liftEitherToAsyncEither(session.verify()));
 
-        const encodedSessionData = this.encodeSession(session);
-
-        try {
-
-            cache.set(session.id, encodedSessionData);
-
-        } catch (err) {
-
-            this._logger.debug(err.message);
-            throw new Error("Error trying to store data in cache.");
-        }
-
-        return session;
     };
 
-    private validateExpiration(expiresIn: number) {
-
-        if (expiresIn <= Date.now()) {
+    public storeSession = (session: Session): EitherAsync<Failure, string> => {
+        const encodedSessionData = Encoding.encodeSession(session);
+        return this.cache.set(session.data[SessionKeys.Id], encodedSessionData);
 
-            this._logger.error(this._sessionExpiredError);
-            throw new Error(this._sessionExpiredError);
-        }
-    }
+    };
 
-    private decodeSession(encodedData: any) {
+    public deleteSession = (cookieString: string): EitherAsync<Failure, number> => {
+        return liftToAsyncEither<Failure, string>(cookieString).chain(this.cache.del);
+    };
 
-        const base64Decoded = Encoding.decodeBase64(encodedData);
+    public load = <T>(cookieId: string): EitherAsync<Failure, T> => {
```

Comment:
Why are load and store public? What is their relationship with `loadSession` and `storeSession` functions?